### ISSUE TITLE
Compatibility fix for WordPress Importer no longer necessary, causing data to be slashed twice

### ISSUE THREAD

**Issue by ocean90** [elementor/elementor#10774]
Originally reported in https://wordpress.org/support/topic/version-0-7-and-elementor-content/.

Since WordPress Importer 0.7 the slashing of meta data has been improved/fixed. This makes the compatibility layer for the post meta `_elementor_data` no longer necessary. It's currently causing the data to be slashed twice.

https://github.com/elementor/elementor/blob/fb9b0812d91ee42e0ab056ce856ffe27bf0ad569/includes/compatibility.php#L319-L368

**Comment by gho5t7m** [comment]
Hi.

Please, fix this issue as soon as possible. I am having the same issue.

Best Regards

**Comment by pollev** [comment]
I can confirm I am also having this issue. I hope a fix can be created soon.

**Comment by zak87** [comment]
Each day, several customers are reporting me the same issue. Hope the fix will be vary soon.

**Comment by simopetrelli** [comment]
I can confirm I am also having this issue. I hope a fix can be created soon.

**Comment by pollev** [comment]
It seems like the fix for this is just to delete the compatibility function calls. Is this issue more complicated than that?

**Comment by DumaBre** [comment]
I have the exact same problem, and it would be really good to solve it as soon as possible.
We want to do full support for Elementor on our Themes but without fixing this bug, our customers will not be able to import Elementor demo content.

**Comment by dusanbre** [comment]
> It seems like the fix for this is just to delete the compatibility function calls. Is this issue more complicated than that?
I tried deleting functions from compatibility, but it doesn't seem to work.

**Comment by a-camacho** [comment]
Hello guys, I think I have the same problem : exporting and importing website as content.xml (with empty site between) is not able to recreate Elementor elements, and leave me with a block of text.

Any information regarding the fix of this issue ?

> It seems like the fix for this is just to delete the compatibility function calls. Is this issue more complicated than that?

Do you know how to do that fix ? I would like to try it out.

**Comment by dusanbre** [comment]
It doesn't work for me, I tried everything, but no ...
Home Page always insert into text block. :(

**Comment by a-camacho** [comment]
Really ? I commented actions in function like this : 

```
	/**
	 * Process post meta before WP importer.
	 *
	 * Normalize Elementor post meta on import, We need the `wp_slash` in order
	 * to avoid the unslashing during the `add_post_meta`.
	 *
	 * Fired by `wp_import_post_meta` filter.
	 *
	 * @since 1.0.0
	 * @access public
	 * @static
	 *
	 * @param array $post_meta Post meta.
	 *
	 * @return array Updated post meta.
	 */
	public static function on_wp_import_post_meta( $post_meta ) {
		
		/*foreach ( $post_meta as &$meta ) {
			if ( '_elementor_data' === $meta['key'] ) {
				$meta['value'] = wp_slash( $meta['value'] );
				break;
			}
		}*/

		return $post_meta;
	}

	/**
	 * Process post meta before WXR importer.
	 *
	 * Normalize Elementor post meta on import with the new WP_importer, We need
	 * the `wp_slash` in order to avoid the unslashing during the `add_post_meta`.
	 *
	 * Fired by `wxr_importer.pre_process.post_meta` filter.
	 *
	 * @since 1.0.0
	 * @access public
	 * @static
	 *
	 * @param array $post_meta Post meta.
	 *
	 * @return array Updated post meta.
	 */
	public static function on_wxr_importer_pre_process_post_meta( $post_meta ) {
		/*
		if ( '_elementor_data' === $post_meta['key'] ) {
			$post_meta['value'] = wp_slash( $post_meta['value'] );
		}
		*/
		return $post_meta;
	}
```

**Comment by zak87** [comment]
I am using WordPress Importer 0.6.4 - but that is just temporary solution. This must be fixed asap.

**Comment by dusanbre** [comment]
Maybe it's because of the plugin version. I'll focus on updating the plugin.
Yes, I know it's a temporary solution, but until the Elementor team resolves this in an update, we have no better solution.:D
Thank you for your help
; D

**Comment by a-camacho** [comment]
> Maybe it's because of the plugin version. I'll focus on updating the plugin.
> Yes, I know it's a temporary solution, but until the Elementor team resolves this in an update, we have no better solution.:D
> Thank you for your help
> ; D

I'm using Wordpress Importer 0.7 and it works with the fix (in case it helps)

**Comment by dusanbre** [comment]
> > Maybe it's because of the plugin version. I'll focus on updating the plugin.
> > Yes, I know it's a temporary solution, but until the Elementor team resolves this in an update, we have no better solution.:D
> > Thank you for your help
> > ; D
> 
> I'm using Wordpress Importer 0.7 and it works with the fix (in case it helps)

Tnx :D

**Comment by dusanbre** [comment]
Yes, it works with a newer version of the plugin, but does anyone have an idea how to extend the Compatibility class to override these 2 methods?

**Comment by wayheming** [comment]
same issue

**Comment by black-eye** [comment]
Hi,

I'm having the same issue even when using importer 0.6.4. Strangely enough, very small number of Elementor pages is imported correctly, on the other hand, all the popups and templates have been imported correctly. Import provided by @zak87 in original WP issue (https://www.dropbox.com/s/696n92k5pes4tb3/ukko.WordPress.2020-01-03.xml?dl=0) seems to work as well.

So I'm probably doing something wrong when exporting the content in first place.

What is the right procedure? It seems you have worked this out somehow. Does it depend on the version of WP or Elementor, presence of Gutenberg or Classic Editor? Do you export the site content completely, or just the pages?

Thanks for any hint.

**Comment by wayheming** [comment]
@black-eye 
wp-content\plugins\elementor\includes\compatibility.php
https://prnt.sc/s7sg2h

try to comment these lines

**Comment by pollev** [comment]
These lines should just be removed from the latest version of Elementor.
It is hilarious that a "compatibility fix" for an older version is now breaking compatibility between the latest versions of both plugins...

**Comment by parlez-web** [comment]
Is there any update on when this will be fixed? Commenting out these lines works, but I have clients asking why the demo import doesn't work. It's frustrating!

**Comment by dusanbre** [comment]
I don't know what the theme structure is, but if it's like in my case, where the wordpress importer is integrated in the theme (it doesn't install everything separately).
Go to the root folder of the importer.
Find file class-wp-import.php
In line 575,802,835 delete the wp_slash_strings_only function

So before deleting it looks like this

add_term_meta ($ term_id, wp_slash ($ key), wp_slash_strings_only ($ value));

add_comment_meta ($ inserted_comments [$ key], wp_slash ($ meta ['key']), wp_slash_strings_only ($ value));

add_post_meta ($ post_id, wp_slash ($ key), wp_slash_strings_only ($ value));

After deleting

add_term_meta ($ term_id, wp_slash ($ key), $ value);
add_comment_meta ($ inserted_comments [$ key], wp_slash ($ meta ['key']), $ value);
add_post_meta ($ post_id, wp_slash ($ key), $ value);

This corrected a mistake in my case.
But this only works if the importer is integrated into the theme.

**Comment by parlez-web** [comment]
Thanks! We haven't integrated the importer in the theme, but found another solution. Removing the filter hooks (put this in our theme) did the trick:

remove_filter( 'wp_import_post_meta', 'Elementor\Compatibility::on_wp_import_post_meta');
remove_filter( 'wxr_importer.pre_process.post_meta', 'Elementor\Compatibility::on_wxr_importer_pre_process_post_meta');

**Comment by dusanbre** [comment]
Nice, that's a good solution ...

**Comment by oooh-boi** [comment]
Oh cmon Elementor guys, do you really need that much time to fix something that simple and make your users happy? It's been over two months...

**Comment by a-camacho** [comment]
> Thanks! We haven't integrated the importer in the theme, but found another solution. Removing the filter hooks (put this in our theme) did the trick:
> 
> remove_filter( 'wp_import_post_meta', 'Elementor\Compatibility::on_wp_import_post_meta');
> remove_filter( 'wxr_importer.pre_process.post_meta', 'Elementor\Compatibility::on_wxr_importer_pre_process_post_meta');

This does not work for me.
Are you using WP Importer (web) or WP CLI Import (cli) ?

I've tried both ways, using these "remove_filter" in my plugin, and it did not work. Did not try in the theme, but it should make no difference no ?

**Comment by parlez-web** [comment]
@a-camacho 
I'm using the normal WP Importer. Hmm... maybe theme hooks are loaded after plugins and Elementor is overriding your code? I'm not sure, I haven't tested it in a plugin setup.
Have you tried adding a priority to the remove_filter hooks?

**Comment by lior-sh** [comment]
Hi @ocean90,

Thanks for reporting.
We will address this issue soon, please follow this thread for updates.
Thanks!

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor v2.9.10

Feel free to update

Thanks!

**Comment by oooh-boi** [comment]
Thanks, shilo-ey for listening to the voice of people and not saying to contact the development team.

**Comment by tanjilahmed7** [comment]
> This issue has been resolved in Elementor v2.9.10
> 
> Feel free to update
> 
> Thanks!

Not Working. I'm still having problems.

**Comment by zak87** [comment]
@tanjilahmed7 All is working fine. You are doing something wrong.
